debian/marquee/nginx/sites/books.spec.whatwg.org.conf

```diff
@@ -0,0 +1,13 @@
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name books.spec.whatwg.org;
+
+    ssl_certificate /etc/letsencrypt/live/books.spec.whatwg.org/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/books.spec.whatwg.org/privkey.pem;
```

Comment:
Could we move these to whatwg.conf and use some kind of templating to determine the file name?

Reply:
ssl_certificate /etc/letsencrypt/live/books.spec.whatwg.org/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/books.spec.whatwg.org/privkey.pem; 

> Could we move these to whatwg.conf and use some kind of templating to determine the file name?

No, because nginx doesn’t itself have any kind of templating mechanism or static macros for use in config files. So I think the only way to do that would be to use a runtime variable in the `ssl_certificate` and `ssl_certificate_key` values—specifically, `$server_name` —if that even worked, which it doesn’t.

To confirm that, I tried putting `ssl_certificate /etc/letsencrypt/live/${server_name}/fullchain.pem` and `ssl_certificate_key /etc/letsencrypt/live/${server_name/privkey.pem` in `whatwg.conf` and removing from all the `sites/*.conf` files. That  caused nginx to fail at startup with `BIO_new_file("/etc/letsencrypt/live/${server_name}/fullchain.pem")`.

Because nginx evaluates that value statically at startup, I guess at that point `${server_name}` is undefined or null, since it’s something that only gets set at runtime.

But even if it worked, I think the problem there is that it would be expensive in that nginx variables are basically all runtime variables that require a lookup per-request.

The nginx docs have an FAQ that discusses the general issue:

https://nginx.org/en/docs/faq/variables_in_config.html

Reply:
Ah, great, thanks for the education!

It does make me think we should perhaps be generating these files using some templating system, but we can do that later, if ever. Not a big deal for now.

Reply:
> It does make me think we should perhaps be generating these files using some templating system

Agreed, that would be better for maintenance

> but we can do that later, if ever. Not a big deal for now.

Yeah, I think once we finally have all the domains moved over, we can go and back and set something up that we can design and test with all cases as a complete set